libyul/AsmAnalysis.cpp

```diff
@@ -156,8 +161,10 @@ vector<YulString> AsmAnalyzer::operator()(Identifier const& _identifier)
 void AsmAnalyzer::operator()(ExpressionStatement const& _statement)
 {
 	vector<YulString> types = std::visit(*this, _statement.expression);
-	if (m_success && !types.empty())
-		typeError(_statement.location,
+	if (success() && !types.empty())
```

Comment:
This is the only place where the `success()`-function is actually needed. At best, it should detect if there was a new error inside `_statement.expression`. Maybe it would be better to add a local variable containing the error count here and remove the `success` function instead?